### ISSUE TITLE
Asking a question

### ISSUE THREAD

**Issue by massatt212** [LukeShortCloud/winesapOS#566]
AMD FSR | Global |
Gamescope | Global

Are these mesa patches or i can enable these with regular mesa?

**Comment by LukeShortCloud** [comment]
Hey, @massatt212 ! These require the use of a global Gamescope session and running SteamOS (or any game launcher, really) with it. SteamOS provides a `gamescope-session` package that is very Steam Deck specific. ChimeraOS provides a more generic package you can use. We'll be officially adding this in our 3.4.0 release of winesapOS. I'm currently working on the 3.3.0 release. For the best results, you will want to use an AMD or Intel graphics device (not NVIDIA).

https://github.com/ChimeraOS/gamescope-session

Related:
- https://github.com/LukeShortCloud/winesapOS/issues/240
- https://github.com/LukeShortCloud/winesapOS/issues/237

Hope this helps!

**Comment by LukeShortCloud** [comment]
Also, here is our current backlog for winesapOS 3.4.0: https://github.com/LukeShortCloud/winesapOS/milestone/13

Expect to see this later this year. :-)

**Comment by massatt212** [comment]
im trying to compile my own mesa, but when i do holoiso and winesape doesnt boot, so im trying to figure out what patches im missing, trying to use fastlinking on holo and winesape

**Comment by LukeShortCloud** [comment]
I'd love to also get that working along with the related GPL feature in Mesa. Let me know if you figure that out. Likewise, I'll do the same and share with you if I can figure it out.

I import the SteamOS source code to a git repository and then host a AUR package to build it. Here is how Mesa is currently built on winesapOS: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=mesa-steamos

**Comment by LukeShortCloud** [comment]
@massatt212 I've got Mesa 23.1 built for testing and can confirm that fast-linking with the new graphics pipeline library (GPL) is amazing with an AMD GPU (it's enabled by default now). I tested a handful of DirectX 9, 10, and 11 games and they all run very smooth now. I'll make sure this gets added into our upcoming winesapOS 3.3.0 release.

I've also been testing around with `gamescope` recently and it's a lot more stable nowadays than it used to be. It (mostly) works on NVIDIA now, too. Integer scaling with `gamescope` works as well. You can set it as launch arguments for games in Steam, Lutris, etc. The Arch wiki has a really good guide on how to use it: https://wiki.archlinux.org/title/Gamescope

**Comment by LukeShortCloud** [comment]
If Mesa 23.2 is not out before our next release, I may also manually enable GPL for Intel graphics, too.

**Comment by massatt212** [comment]
You should focus on steamos3 clone with latest mesa and kernels, many people want that, I'm sure you would be featured in lots of videos.

**Comment by LukeShortCloud** [comment]
Hey @massatt212 , thanks for the feedback! You have helped me rethink how I approach winesapOS. I just released winesapOS 3.3.0 which is a step in this direction.

For now, I have made sure to enable GPL support for AMD, Intel, and NVIDIA graphics devices in this new release. Feel free to try it out and let me know if that works for you!

https://github.com/LukeShortCloud/winesapOS/releases/tag/3.3.0
